### PR TITLE
Signup login implementation

### PR DESCRIPTION
**Pull request by marinPh**
New error and crash prevention in Signing up

### CODE REVIEW

File: app/src/main/java/com/swent/assos/MainActivity.kt
Comment (Tikipiou) [inline app/src/main/java/com/swent/assos/MainActivity.kt:1]: Move this in `Config.kt`.

File: app/src/main/java/com/swent/assos/model/view/LoginViewModel.kt
Comment (Tikipiou) [inline app/src/main/java/com/swent/assos/model/view/LoginViewModel.kt:1]: It's better to have two callbacks : `onSuccess` and `onError`

**Review by Tikipiou — CHANGES_REQUESTED** [review]
Overall, this is good work.

File: app/src/main/java/com/swent/assos/model/view/LoginViewModel.kt
Comment (marinPh) [reply app/src/main/java/com/swent/assos/model/view/LoginViewModel.kt:1]: Would influence behaviour of the tests

**Review by Tikipiou — APPROVED** [review]
Very good job !

**Pull request merged**